src/api-service/__app__/timer_retention/__init__.py

```diff
@@ -0,0 +1,87 @@
+import datetime
+import logging
+
+import azure.functions as func
+from onefuzztypes.enums import JobState, TaskState
+
+from ..onefuzzlib.events import get_events
+from ..onefuzzlib.jobs import Job
+from ..onefuzzlib.notifications.main import Notification
+from ..onefuzzlib.repro import Repro
+from ..onefuzzlib.tasks.main import Task
+
+RETENTION_POLICY = datetime.timedelta(days=(18 * 30))
+SEARCH_EXTENT = datetime.timedelta(days=(20 * 30))
+
+
+def main(mytimer1: func.TimerRequest, dashboard: func.Out[str]) -> None:  # noqa: F841
+
+    now = datetime.datetime.now(tz=datetime.timezone.utc)
+
+    time_retained_older = now - RETENTION_POLICY
+    time_retained_newer = now - SEARCH_EXTENT
+
+    time_filter = (
+        f"Timestamp lt datetime'{time_retained_older.isoformat()}' "
+        f"and Timestamp gt datetime'{time_retained_newer.isoformat()}'"
+    )
+    time_filter_newer = f"Timestamp gt datetime'{time_retained_older.isoformat()}'"
+
+    # Collecting 'still relevant' task containers.
+    used_containers = set()
+    for task in Task.search(raw_unchecked_filter=time_filter_newer):
+        task_containers = {x.name for x in task.config.containers}
+        used_containers.update(task_containers)
+
+    # You have to do notification before task,
+    # because editing the upn for tasks will change the timestamp
+    for notification in Notification.search(raw_unchecked_filter=time_filter):
+        logging.info(

```

Comment:
Assuming a notification is old, but still in use, this will get logged once a day forever.  This makes it a debug log, which won't get logged to app insights by default.

```suggestion
        logging.debug(
```